bloom-housing/bloom
# 1480/ Fix double submit

# Pull Request Template

## Issue
#1480 

Addresses # (issue)

## Description

This is a resubmission of the PR I had out yesterday, but with updates to address potential race conditions. I didn't want to just void out the onSubmit function on the form, though harmless, it would create a detached form warning. triggerSubmit now calls a setState hook instead of onSubmit directly and there is now a useEffect hook that looks for submitData and status, so when they're both valid, the form submits.

## Type of change


## How Can This Be Tested/Reviewed?
Edit or create a new listing, submit should not be firing twice and there should be no warning about the form being detached.


## Checklist:
